sdcm/cluster.py

```diff
@@ -605,6 +609,39 @@ def cpu_cores(self):
         cores = int(cores) if cores else None
         return cores
 
+    @property
+    def scylla_shards(self):
+        if not self._scylla_shards:
+            # Possible output of grep:
+            #   'CPUSET="--cpuset 1-7 "'
+            #   'CPUSET="--cpuset 1-7,9-15 "'
+            grep_result = self.remoter.run('grep "^CPUSET" /etc/scylla.d/cpuset.conf')
+            scylla_cpu_set = re.findall(r'(\d+)', grep_result.stdout)
+            self.log.info(f"grep_result: {grep_result}")
+
+            if not scylla_cpu_set:
+                self.log.error(f'Failed to parse CPUSET from "/etc/scylla.d/cpuset.conf". '
+                               f'Unexpected CPUSET is "{grep_result}"')
+                self._scylla_shards = self.cpu_cores
+
+            self._scylla_shards = int(scylla_cpu_set[1]) - int(scylla_cpu_set[0]) + 1
+
+            if len(scylla_cpu_set) == 4:
```

Comment:
Why there is a special treatment to 4?

Reply:
Or it's just means that the lens of the list is 4 and we got splitted ranges?
Anyway, worth a comment to explain it.

Reply:
I put expected string, that is analysing here, a few rows above. It show what we are going to analyse